recipes/default.rb

```diff
@@ -18,23 +18,37 @@
 #
 include_recipe 'chef-sugar'
 
-package 'apt-transport-https' if ubuntu?
+# setup docker apt repo
+include_recipe 'chef-apt-docker'
 
-apt_repository 'docker' do
-  uri 'https://apt.dockerproject.org/repo'
-  trusted true
-  components ["ubuntu-#{node['lsb']['codename']}", 'main']
-  only_if { ubuntu? }
+# IAM role support for tasks
+include_recipe 'sysctl::default'
+include_recipe 'iptables::default'
+sysctl_param 'net.ipv4.conf.all.route_localnet' do
+  value 1
+end
+iptables_rule 'dnat_80_to_51679' do
```

Comment:
These iptables rules create a notification `notifies :run, 'execute[rebuild-iptables]', :delayed`
Meanwhile, docker-engine gets installed and adds it's own iptables rules
At the end of chef run, iptables get rebuilt with files from `/etc/iptables.d/` and Docker's rules from NAT table disappear. And Docker cannot create containers which have port bindings.
Not sure about how to solve this 🤔